### ISSUE TITLE
Schedule - Time Range Description Replaced with URL Encoded String

### ISSUE THREAD

**Issue by kuya1284** [opnsense/core#6348]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

When saving a Time Range for a Schedule, the Time Range Description gets replaced with a URL encoded string.  For example, when I attempt to save the description as "9:00 PM to 11:59 PM", the string gets saved and replaced with "9%3A00%20PM%20to%2011%3A59%20PM", making the description very difficult to read. I'm new to OPNsense and first noticed this issue with 22.7.

**To Reproduce**

1. Go to **Firewall > Settings > Schedule** (https://[OPNsense IP]/firewall_schedule.php)
2. Click on the **Add** button (or edit an existing entry)
3. Configure the time range 
4. In **Time Range Description**, enter a description containing characters like spaces, colons, etc.
5. Click the **Add Time** button
6. The new entry will appear on the **Configured Ranges** row with the **Time Range Description** re-saved as a URL encoded string
7. Click the **Save** button
8. The new schedule will appear on the list of schedule entries, along with the **Time Range Description** as a URL encoded string

**Expected behavior**

The **Time Range Description** should appear just as I entered it. If a URL encoded string needs to be saved on the backend, that's fine. But when rendering the description, it must be URL decoded to not only make the description more readable, but also to prevent double-encoding in case the description were to be re-saved.

**Describe alternatives you considered**

I have no known workarounds or alternatives.

**Screenshots**

![image](https://user-images.githubusercontent.com/121985/220165404-87211c15-3a83-41fd-8eff-b4ca1bbd88fa.png)
![image](https://user-images.githubusercontent.com/121985/220165466-f1395727-d5bc-429c-9bf6-a25f82b93203.png)

**Relevant log files**

N/A

**Additional context**

N/A

**Environment**

OPNsense 23.1.1_2 (amd64, OpenSSL)
Protectli VP2410
Intel® Celeron® J4125 2.0GHz Quad Core

**Comment by AdSchellevis** [comment]
It's unfortunately a known omission with very low priority due to the low usage and complexity of the legacy page, more info in https://github.com/opnsense/core/issues/3357

**Comment by kuya1284** [comment]
> It's unfortunately a known omission with very low priority due to the low usage and complexity of the legacy page, more info in https://github.com/opnsense/core/issues/3357

Thanks, @AdSchellevis. I tried to find any existing issues to avoid reporting a duplicate, but failed in my search. Sorry about that and thanks for pointing me to the previously reported issue.

**Comment by AdSchellevis** [comment]
@kuya1284 no problem, this one is pretty hard to find. At some point a rewrite of the whole schedule idea would be better, but given the investment required and the low business interest I don't expect that to happen any time soon.

**Comment by kuya1284** [comment]
@AdSchellevis , would you mind if I take a stab at this, as well as https://github.com/opnsense/core/issues/6349? 

I started tinkering around a bit and this seems like it would be a fun mini project. As a matter of fact, I think it'd be fun to improve some of the UX as well.

![image](https://user-images.githubusercontent.com/121985/220867702-5773f73a-1276-4c2c-ae67-f1d215083951.png)

I have some ideas in mind on how to improve this, but if you and your team had a different direction, I'm all ears. ;)

Thanks.

**Comment by AdSchellevis** [comment]
@kuya1284 to be honest, I don't really have ideas about this. Ideally we should try to migrate it to simple MVC forms at some point, but an overhaul of the screen and improving the code quality would be acceptable as well in my opinion.

**Comment by kuya1284** [comment]
I just wanted to give an update that this issue has been fixed. I have the fix in my fork and just need to submit the PR. I also fixed #6349. 

I already began working on some improvements in a new branch to hopefully help with things that some people have requested in the past, like a more user-friendly way of handling repeating time ranges, as well as the ability to specify and end date/time.

Should I create a feature request so that I could reference the new issue with my commits? Or is it ok to make commits without having a feature request? I wasn't sure how things like that get handled in this project.
